compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5962,6 +5963,163 @@ OMR::X86::TreeEvaluator::bitpermuteEvaluator(TR::Node *node, TR::CodeGenerator *
+      TR::MemoryReference *srcMR = generateX86MemoryReference(srcNode, cg);
+      srcReg = cg->allocateRegister(TR_GPR);
+      generateRegMemInstruction(extendOpCode, node, srcReg, srcMR, cg);
+      resReg = srcReg;
+      }
+   // If no zero-extension shortcut applied, default to zero-extending the src
+   if (resReg == NULL)
+      {
+      TR::InstOpCode::Mnemonic extendOpCode = isShort ? TR::InstOpCode::MOVZXReg4Reg2 : TR::InstOpCode::MOVZXReg4Reg1;
+      srcReg = cg->gprClobberEvaluate(srcNode, extendOpCode);
+      resReg = srcReg;
+      }
+   // Evaluate the missing operand
+   if (maskReg == NULL)
+      {
+      if (maskNode->getReferenceCount() == 1 && maskNode->getRegister() == NULL && maskNode->getOpCode().isLoadVar())
+         maskMR = generateX86MemoryReference(maskNode, cg);
+      else
+         maskReg = cg->evaluate(maskNode);
+      }
+   else
+      {
+      srcReg = cg->evaluate(srcNode);
+      }
+
+   if (maskMR)
+      {
+      generateRegRegMemInstruction(TR::InstOpCode::PEXT4RegRegMem, node, resReg, srcReg, maskMR, cg);
+      }
+   else
+      {
+      generateRegRegRegInstruction(TR::InstOpCode::PEXT4RegRegReg, node, resReg, srcReg, maskReg, cg);
+      }
+
+   node->setRegister(resReg);
+
+   if (maskMR)
+      maskMR->decNodeReferenceCounts(cg);
+   else
+      cg->decReferenceCount(maskNode);
+
+   cg->decReferenceCount(srcNode);
+
+   return resReg;
+   }
+
+TR::Register*
+OMR::X86::TreeEvaluator::bcompressbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return smallCompressBitsEvaluator(node, cg);
+   }
+
+TR::Register*
+OMR::X86::TreeEvaluator::scompressbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return smallCompressBitsEvaluator(node, cg);
+   }
+
+TR::Register*
+OMR::X86::TreeEvaluator::icompressbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return compressExpandBitsEvaluator(node, cg, TR::InstOpCode::PEXT4RegRegReg, TR::InstOpCode::PEXT4RegRegMem);
+   }
+
+TR::Register*
+OMR::X86::TreeEvaluator::lcompressbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return compressExpandBitsEvaluator(node, cg, TR::InstOpCode::PEXT8RegRegReg, TR::InstOpCode::PEXT8RegRegMem);
+   }
+
+TR::Register*
+OMR::X86::TreeEvaluator::bexpandbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return iexpandbitsEvaluator(node, cg);
+   }
+
+TR::Register*
+OMR::X86::TreeEvaluator::sexpandbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return iexpandbitsEvaluator(node, cg);
+   }
+
+TR::Register*
+OMR::X86::TreeEvaluator::iexpandbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return compressExpandBitsEvaluator(node, cg, TR::InstOpCode::PDEP4RegRegReg, TR::InstOpCode::PDEP4RegRegMem);
+   }
+
+TR::Register*
+OMR::X86::TreeEvaluator::lexpandbitsEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   return compressExpandBitsEvaluator(node, cg, TR::InstOpCode::PDEP8RegRegReg, TR::InstOpCode::PDEP8RegRegMem);
+   }
```

Comment:
Sorry for catching this late, but this evaluator should live in `amd64` (not common code) because it uses an instruction that is only available on 64-bit, and you've already introduced a special evaluator in `i386`.